include/ginkgo/core/reorder/rcm.hpp

```diff
@@ -224,6 +164,92 @@ class Rcm : public EnablePolymorphicObject<Rcm<ValueType, IndexType>,
 
 
 }  // namespace reorder
+
+
+namespace experimental {
+namespace reorder {
+
+
+using rcm_starting_strategy = gko::reorder::starting_strategy;
+
+
+/**
+ * Rcm is a reordering algorithm minimizing the bandwidth of a matrix. Such a
+ * reordering typically also significantly reduces fill-in, though usually not
+ * as effective as more complex algorithms, specifically AMD and nested
+ * dissection schemes. The advantage of this algorithm is its low runtime.
+ *
+ * The class is a LinOpFactory generating a Permutation matrix out of a Csr
+ * system matrix, to be used with `Csr::permute(...)`.
+ *
+ * There are two "starting strategies" currently available: minimum degree and
+ * pseudo-peripheral. These strategies control how a starting vertex for a
+ * connected component is chosen, which is then renumbered as first vertex in
+ * the component, starting the algorithm from there.
+ * In general, the bandwidths obtained by choosing a pseudo-peripheral vertex
+ * are slightly smaller than those obtained from choosing a vertex of minimum
+ * degree. On the other hand, this strategy is much more expensive, relatively.
+ * The algorithm for finding a pseudo-peripheral vertex as
+ * described in "Computer Solution of Sparse Linear Systems" (George, Liu, Ng,
+ * Oak Ridge National Laboratory, 1994) is implemented here.
```

Comment:
I think you should mention that the algorithm needs a symmetric matrix and that we have version that can explicitly symmetrize the matrix.